### ISSUE TITLE
Support Python 3.12 & 3.13 - was "imghdr module not found"

### ISSUE THREAD

**Issue by Mukei** [MartinPacker/md2pptx#156]
After installing python-pptx and cloning the directory, I tried running my first markdown file conversion.
Unfortunately, I ran into the below error.

**Input**
```bash
python3 -m md2pptx ./Downloads/output.pptx < ./Downloads/202404301018\ -\ Test.md
```

**Output**
```bash
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/XX/Prog/01-Projects/md2pptx/md2pptx.py", line 28, in <module>
    import imghdr, struct
ModuleNotFoundError: No module named 'imghdr'
```

Did I miss any steps?

I am using python 3.13, on OSX 14.3 (23D56).

**Comment by MartinPacker** [comment]
Hi @Mukei. First, thank you for your bug report. I take currency seriously - but haven't seen benefits in 3.12 or 3.13 for this project. Nevertheless it needs to run.

I discovered an issue with 3.12 - which I aim to get a fix out for soon. Another deprecated behaviour going away, of course.

I actually had trouble getting a 3.12 install to find python-pptx  - but obviously I need to fix that.

What I think the current use of `imghdr` is to recognise the dimensions (really proportions) of an image and its type. I'll have to see how I can replicate that function - preferably not creating another dependency (that might go away).

Anyhow, please do test on 3.12. I think you'll find it breaks there, too.

**Comment by MartinPacker** [comment]
It looks like Pillow at a high enough level could do. (9.5.0 for 3.12?) We already make it an optional prereq. Probably have to make it a mandatory one.

**Comment by MartinPacker** [comment]
I was wrong. imghdr is apparently just picking up the filetype for a graphic. Will have to think about how to do that differently. (Pillow is used for something else and can remain optional.)

**Comment by MartinPacker** [comment]
[This Stack Overflow article](https://stackoverflow.com/questions/889333/how-to-check-if-a-file-is-a-valid-image-file) poses the idea that one could use the file extension:

```
filename.lower().endswith(('.png', '.jpg', '.jpeg', '.tiff', '.bmp', '.gif'))
```

I'm not wowed by this but maybe imghdr was doing little more than that.

**Comment by MartinPacker** [comment]
Built not using imghdr but testing filename extensions. Please try the latest md2pptx file out. On 3.12 if you like. Or maybe 3.13. Let me know how it goes. I want to push a release that says it supports 3.12.

**Comment by MartinPacker** [comment]
My initial tests with 3.12.2 and 3.13.0a5 show md2pptx continuing to work. (Installed both via pyenv and installed python-pptx into each environment.)

Thanks to your opening this issue @Mukei I now know how to test with newer versions. That should help keep this (and [mdpre](https://github.com/MartinPacker/md2pptx) current.

**Comment by MartinPacker** [comment]
Fixed and shipped in [v4.4](https://github.com/MartinPacker/md2pptx/releases/tag/v4.4).

Closing. Please reopen if you don't consider it fixed.
